src/react/projects/spark-react/src/SprkStepper/SprkStepper.js

```diff
@@ -0,0 +1,247 @@
+    const { activeStepIndex } = this.state;
+
+    let hasCarousel = false;
+    let hasDesc = false;
+    let descCount = 0;
+
+    children.forEach((child, index) => {
+      if (child.props.imgSrc) {
+        // TODO validate that if one has it they all have it
+        // hasCarousel = true;
+      }
+
+      if (child.props.children) {
+        descCount++;
+      }
+    });
+
+    if (descCount === children.length) {
+      hasDesc = true;
+    } else if (descCount > 0){
+      // true if 1 or more has a description but not all
+      console.warn("SprkStepper: If any step has a description, they all must have a description.");
+    }
+
+    // build the carousel steps as needed
+    const carousel = (
+      <SprkCarousel>
+        {children.map((childNode, index) => {
+          return (
+            <SprkCarouselStep
+              key={index} //TODO lodash or something
+              imgSrc={childNode.props.imgSrc}
+              imgAlt={childNode.props.imgAlt}
+              isSelected={activeStepIndex === index}
+            >
+
+            </SprkCarouselStep>
+          )
+        })}
+      </SprkCarousel>
+    );
+
+    let stepper = (
+      <ol
+        className={classnames(
+          'sprk-c-Stepper',
+          hasDarkBackground ? 'sprk-c-Stepper--has-dark-bg' : '',
+          additionalClasses
+        )}
+        data-sprk-stepper="container"
+        role="tablist"
+        aria-orientation="vertical" // todo change based on window width
+        data-id={idString}
+        {...other}
+      >
+        {hasDesc &&
+          <SprkStepperSlider
+            title={children[activeStepIndex].props.title} // active title
+            contents={children[activeStepIndex].props.children} // active contents
+            sliderStyle={this.state.sliderStyle}
+          />
+        }
+
+        {children.map((childNode, index) => {
+
+          let step = (
+            <SprkStepperStep
+              {...childNode.props}
+              key={index}
+              id={childNode.id}
+              isSelected={activeStepIndex === index}
+              tabIndex={0}
+              additionalClasses={classnames(
+                index === 0 ? 'sprk-c-Stepper__step--first' : '',
+                index === children.length - 1 ? 'sprk-c-Stepper__step--last' : '',
+                childNode.props.additionalClasses
+              )}
+              onKeyDown={this.handleKeyEvents}
+              onClick={(event) => { childNode.props.onClick(); this.handleStepClick(event, index); }}
+              renderCallback={ this.updateSliderPosition }
+              hasDescription={hasDesc}
+              hasDarkBackground={hasDarkBackground}
+            />
+          );
+
+          return step;
+        })}
+      </ol>
+    );
+
+    if (hasDarkBackground) {
+      stepper = (
+        <div className='sprk-u-BackgroundColor--blue sprk-o-Box sprk-o-Box--large'>
```

Comment:
could these conditionally added classes be included in the `classnames` of stepper?

Reply:
@bobwise  did you address this feedback with RV yet?